GPSTest/src/main/java/com/chesapeaketechnology/gnssmonkey/GpsMonkeyActivity.java

```diff
@@ -182,6 +159,7 @@ protected void onStop() {
         if (serviceBound) {
             try {
                 unbindService(serviceConnection);
+                serviceBound = false;
```

Comment:
It is hard to tell in this pull request, but it looks like serviceBound is being accessed from multiple threads, so we probably need to make this an AtomicBoolean, or at a minimum volatile depending on what thread(s) update it.